tests/test_event_detection.py

```diff
@@ -1,7 +1,180 @@
+'''Code Tests'''
+
+# Authors: Darcy Diesburg <darcy.diesburg@gmail.com>
+
+import sys
+import os
+import os.path as op
+
 import pytest
 import numpy as np
 from scipy.io import loadmat
 
+sys.path.append('/gpfs/home/ddiesbur/SpectralEvents')
+import spectralevents as se
+
+def test_event_comparison(method):
+    '''Test whether the output from MATLAB and Python event detection scripts align
+
+    Parameters
+    ----------
+    method : float
+    The event detection method (1, 2, or 3) used to detect spectral events.
+
+    Notes
+    -----
+    Currently only supports testing method 1 and 3, with method 2 to be added
+    to the Python event detection functions in the future. Python output is
+    tested against the output of MATLAB scripts from Shin et al., 2017. Detection
+    of the same number of beta events within each trial is required, and tolerable
+    margins of deviation for the other parameters.
+    '''
+
+    # Define tolerable deviance for spectral event characteristics
+    dev_max_time = 5  # +-5ms
+    dev_max_freq = 1  # +-1Hz
+    dev_low_freq = 1  # +-1Hz
+    dev_high_freq = 1  # +-1Hz
+    dev_onset = 10  # +-10ms
+    dev_offset = 10  # +-10ms
+    dev_fom_pow = 0.5  # +-0.5 FOM
+
+    # Load MATLAB data
+    data_dir = os.getcwd()
+    if method == 1:
+        matlab_file = loadmat(
+            op.join(data_dir, 'tests', 'beta_events_shin_2017.mat'))
+    #elif method == 3:
+    #    matlab_file = loadmat(
+    #        op.join(data_dir, 'tests', 'beta_events_shin_2017_method3.mat'))
+    else:
+        raise ValueError('Unsupported method! Please use 1') #or 3')
+    mat_all_evs = matlab_file['event_times'][:, 0]
+
+    # Get python events by running find_events on demo data
+    # <----------SIMPLIFY BY GETTING PRE-SAVED PICKLE FILES FOR THIS INSTEAD----------->
```

Comment:
We actually want it to run the whole analysis pipeline every time. Am I misunderstanding what you're suggesting here?

Reply:
Nope, this was a misunderstanding on my part of what we were going to want it to do, so I'll remove this placeholder.